core/src/main/scala/org/bitcoins/core/protocol/ln/LnInvoiceTags.scala

```diff
@@ -1,98 +1,129 @@
 package org.bitcoins.core.protocol.ln
 
-import org.bitcoins.core.crypto.Sha256Digest
-import org.bitcoins.core.protocol.{ Bech32Address, BitcoinAddress }
-import org.bitcoins.core.util.BitcoinSUtil
+import org.bitcoins.core.crypto.{ ECPublicKey, Sha256Digest }
 import org.bitcoins.core.number.{ UInt64, UInt8 }
+import org.bitcoins.core.protocol._
+import org.bitcoins.core.protocol.ln.LnInvoiceTag.PaymentHashTag
 import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.core.util.Bech32
+import org.slf4j.LoggerFactory
+import scodec.bits.ByteVector
 
-sealed abstract class LnInvoiceTags {
-  def paymentHash: Sha256Digest
+sealed abstract class LnInvoiceTag extends NetworkElement {
 
-  def description: Option[String]
+  def prefix: LnTagPrefix
 
-  def spk: Option[ScriptPubKey]
+  /** Converts the data portion of the LnInvoiceTag to a bech32 string */
+  def dataToBech32: String = {
+    Bech32.encode8bitToString(bytes)
+  }
 
-  def descriptionHash: Option[Sha256Digest]
+  override def toString: String = {
```

Comment:
I like splitting out the Tags into their own objects, and giving each their own serialization / `toString` capabilities, but as far as I can tell, none of the encoding used here is correct. We will need to go back and revise.